core/frontend/src/Viewport.ts

```diff
@@ -1353,17 +1379,37 @@ export abstract class Viewport implements IDisposable {
       provider.forEachTileTreeRef(this, (ref) => func(ref));
   }
 
+  /** @internal */
+  public forEachMapTreeRef(func: (ref: TileTreeReference) => void): void {
+    if (this._mapTiledGraphicsProvider)
+      this._mapTiledGraphicsProvider.forEachTileTreeRef(this, (ref) => func(ref));
+  }
+
   /** @internal */
   public forEachTileTreeRef(func: (ref: TileTreeReference) => void): void {
     this.view.forEachTileTreeRef(func);
     this.forEachTiledGraphicsProviderTree(func);
+    this.forEachMapTreeRef(func);
+  }
+
+  /**
+   * Returns true if all [[TileTree]]s required by this viewport have been loaded.
+   */
+  public get areAllTileTreesLoaded(): boolean {

```

Comment:
This needs to invoke `this.view.areAllTileTreesLoaded`. It should then check all tile tree refs owned by the viewport (and not those owned by the ViewState).
See specializations of `areAllTileTreesLoaded` by SheetViewState and DrawingViewState.

Reply:
Pushed this change.
In a separate PR I think we should replace all the `forEachXXXRef` functions with `Iterable<TileTreeReference>`s - they would be less ugly and allow you to `break`.